pkg/grafana/client_test.go

```diff
@@ -0,0 +1,162 @@
+package grafana
+
+import (
+	"testing"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/kanopy-platform/grafana-auth-proxy/internal/config"
+	"github.com/stretchr/testify/assert"
+)
+
+func setupUser() gapi.User {
+	return gapi.User{
+		ID:    1,
+		Email: "foo@example.com",
+		Login: "foo",
+	}
+}
+
+func TestLookupUser(t *testing.T) {
+	user := setupUser()
+
+	client := NewMockClient(user, nil)
+
+	foundUser, err := client.LookupUser(user.Login)
+	assert.Nil(t, err)
+	assert.Equal(t, user.Login, foundUser.Login)
+
+	notFoundUser, err := client.LookupUser("")
+	assert.Nil(t, err)
+	assert.Equal(t, gapi.User{}, notFoundUser)
+}
+
+func TestCreateUser(t *testing.T) {
+	user := setupUser()
+
+	client := NewMockClient(user, nil)
+
+	uid, err := client.CreateUser(user)
+	assert.Nil(t, err)
+	assert.Equal(t, uid, user.ID)
+}
+
+func TestAddOrgUser(t *testing.T) {
+	user := setupUser()
+
+	orgRoleMap := userOrgsRoleMap{
+		1: models.ROLE_EDITOR,
+	}
+	client := NewMockClient(user, orgRoleMap)
+
+	// test adding to new org
+	err := client.AddOrgUser(2, "foo", "Editor")
+	assert.NoError(t, err)
+
+	// test already a member
+	err = client.AddOrgUser(1, "foo", "Editor")
+	assert.Contains(t, err.Error(), "User is already member")
+}
+
+func TestUpsertOrgUser(t *testing.T) {
```

Comment:
This is missing two test cases:
 * grafanaAdmin
 * IsRoleAssignable 
 
 
 With the addition of these, it may be worth updating this to a table test format.

Reply:
agreed, I think other tests will also be better using table testing